src/main/scala/diplomaticobjectmodel/DiplomaticObjectModelUtils.scala

```diff
@@ -186,4 +186,34 @@ object DiplomaticObjectModelAddressing {
         writeMaskGranularity = granWidth
       )
     }
+
+  private def getInterruptNumber(r: ResourceValue): BigInt = {
+    r match {
+      case ResourceInt(value: BigInt) => value
+      case _ => throw new IllegalArgumentException
+    }
+  }
+
+  private def getDeviceName(device: Device): String = {
+    device match {
+      case sd:SimpleDevice => sd.asInstanceOf[SimpleDevice].devNamePlusAddress
+      case _ => throw new IllegalArgumentException
+    }
+  }
+
+  def describeInterrupts(name: String, resources: ResourceBindings): Seq[OMInterrupt] = {
+    val int = resources("int")
+    int.flatMap {
+      b =>
+        val grandParentOpt = b.device.get.parent
+        grandParentOpt.map {
+         gp =>
+          OMInterrupt(
+            receiver = getDeviceName(gp),
+            numberAtReceiver = getInterruptNumber(b.value).toInt,
```

Comment:
Maybe `numberAtReceiver` should be a BigInt instead of you casting it to an int?

Reply:
Sure

Reply:
There are DiplomaticObjectModelUtils functions which receive params objects which need their signature refactored.